waltz-ng/client/data-types/resolvers.js

```diff
@@ -0,0 +1,26 @@
+export function loadDataTypes(dataTypeService) {
+    return dataTypeService.loadDataTypes();
+}
+
+loadDataTypes.$inject = ['DataTypeService'];
+
+
+export function dataTypeIdResolver($q, $stateParams, dataTypeService) {
```

Comment:
Maybe I've got this all wrong... But why not have the resolver return the data type entity not the id. Then have two resolvers 
, one for byCode and one for byId,  both should be simple proxies to the service and would remove need for the $q service ?